src/vs/workbench/services/search/common/textSearchManager.ts

```diff
@@ -31,7 +31,7 @@ export class TextSearchManager {
 	search(onProgress: (matches: IFileMatch[]) => void, token: CancellationToken): Promise<ISearchCompleteStats> {
 		const folderQueries = this.query.folderQueries || [];
 		const tokenSource = new CancellationTokenSource();
-		token.onCancellationRequested(() => tokenSource.cancel());
+		const listener = token.onCancellationRequested(() => tokenSource.cancel());
```

Comment:
You should be able to just do

```
const tokenSource = new CancellationTokenSource(token);
```

(GH won't let me add a suggestion for both lines apparenty)